client/state/jetpack-connect/actions.js

```diff
@@ -196,32 +196,72 @@ export function retryAuth( url, attemptNumber ) {
 	};
 }
 
-export function createAccount( userData ) {
+export function createAccount( userData, { service, access_token, id_token } = {} ) {
 	return dispatch => {
 		dispatch( recordTracksEvent( 'calypso_jpc_create_account', {} ) );
 
 		dispatch( {
 			type: JETPACK_CONNECT_CREATE_ACCOUNT,
-			userData: userData,
+			userData,
 		} );
-		wpcom.undocumented().usersNew( userData, ( error, data ) => {
-			if ( error ) {
-				dispatch(
-					recordTracksEvent( 'calypso_jpc_create_account_error', {
-						error_code: error.code,
-						error: JSON.stringify( error ),
-					} )
-				);
-			} else {
-				dispatch( recordTracksEvent( 'calypso_jpc_create_account_success', {} ) );
-			}
-			dispatch( {
-				type: JETPACK_CONNECT_CREATE_ACCOUNT_RECEIVE,
-				userData: userData,
-				data: data,
-				error: error,
+
+		if ( service ) {
```

Comment:
This is shamelessly ripped off from https://github.com/Automattic/wp-calypso/blob/663db1df43c819b7ed0a7a58733fa11a38e2f3b5/client/lib/signup/step-actions.js#L331-L339

Reply:
Could we centralize this and reuse it instead of duplicating the code?

Reply:
I was trying to walk a fine line with this. Technically, we should be able to `import { createAccount } from 'lib/signup/step-actions'` and call it, but we'd be relying heavily on the internals of that implementation.

I don't think there's a lot of value in abstracting this code which doesn't follow current best practices.

If we'd like to clean this up, I think the largest benefit would be migrating some of the user creation bits to the data-layer. That implementation could then be used here and signup could migrate in the future. For the moment I've avoided that because this seems to work fine and the data-layer work would require several PRs to get everything in place.

I'm certainly happy to explore these options more.